kratos/solving_strategies/schemes/residual_based_bdf_scheme.h

```diff
@@ -383,9 +327,11 @@ class ResidualBasedBDFScheme
         std::vector< Vector > dot2un0; /// Second derivative
     };
 
-    const std::size_t mOrder; /// The integration order
-    Vector mBDF;              /// The BDF coefficients
-    GeneralVectors mVector;   /// The structure containing the  derivatives
+    const std::size_t mOrder;                      /// The integration order
+    Vector mBDF;                                   /// The BDF coefficients
+    GeneralVectors mVector;                        /// The structure containing the  derivatives
+    TimeDiscretization::BDF::Pointer mpBDFUtility; /// Utility to compute BDF coefficients
```

Comment:
I don't see the need for this to be a pointer

Reply:
you are right, the habit... i will change it

Reply:
I think it needs to be a pointer. Time discretization does not have copy constructor (it has a unique pointer inside) and time scheme class needs it.

Reply:
And if you make it a pointer we can use any BDF order (1,2,3..)

Reply:
ok I get it now :)

Reply:
btw does this even compile?
I thought you changed to unique ptr?

Reply:
My bad!, I think it is actually still a shared pointer. If I change this to:

`Kratos::unique_ptr<TimeDiscretization::BDF> mpBDFUtility;`

I get an error in the copy constructor, so I will leave it as a shared pointer (using make_shared)

Reply:
ok